app/searchers/responses_searcher.rb

```diff
@@ -131,6 +131,14 @@ def clean_up
     self.form_ids = form_ids.uniq
     self.submitters = submitters.uniq
     self.groups = groups.uniq
+    # Map possible_ids to a single id based on any active form filters.
+    self.qings = qings.map do |qing|
+      return qing unless qing[:possible_ids]
+      qings = Questioning.where(id: qing[:possible_ids])
+      qings = qings.where(form_id: form_ids) if form_ids.present?
+      qing[:id] = qings.filter_unique.pluck(:id).first
+      qing.except(:possible_ids)
```

Comment:
This block is a bit tough to read. A few things: 

1. You are shadowing `self.qings`—perhaps choose a different var name for inside the loop
2. The block feels like it should be its own method with a nice descriptive name. A comment on the block like this is often a hint that a method is called for.
3. What was once a `clean_up` method is really now a second parsing pass. I would call it that perhaps. Or maybe rename `clean_up` to `dedupe` or something and then call that and then the new method all from `save_filter_data`. Or whatever feels right to you. But the narrowing of the question options feels like more than just cleaning up.